packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -12,95 +13,242 @@ import {
   UseFieldForwardedProps,
   UseFieldInternalProps,
   AvailableAdjustKeyCode,
+  FieldValueManager,
 } from './useField.interfaces';
 import {
-  cleanTrailingZeroInNumericSectionValue,
   getMonthsMatchingQuery,
   getSectionValueNumericBoundaries,
   getSectionVisibleValue,
   adjustDateSectionValue,
   adjustInvalidDateSectionValue,
   setSectionValue,
+  applySectionValueToDate,
+  createDateFromSections,
+  cleanTrailingZeroInNumericSectionValue,
 } from './useField.utils';
 
-export const useField = <
-  TInputValue,
-  TValue,
-  TDate,
-  TSection extends FieldSection,
-  TForwardedProps extends UseFieldForwardedProps,
-  TInternalProps extends UseFieldInternalProps<any, any, any>,
->(
-  params: UseFieldParams<TInputValue, TValue, TDate, TSection, TForwardedProps, TInternalProps>,
-): UseFieldResponse<TForwardedProps> => {
+const useFieldState = <TValue, TDate, TSection extends FieldSection>({
+  fieldValueManager,
+  valueManager,
+  value: valueProp,
+  defaultValue,
+  onChange,
+  format,
+}: {
+  valueManager: PickerStateValueManager<TValue, TValue, TDate>;
+  fieldValueManager: FieldValueManager<TValue, TDate, TSection, any>;
+  defaultValue: TValue | undefined;
+  value: TValue | undefined;
+  onChange?: (value: TValue) => void;
+  format: string;
+}) => {
   const utils = useUtils<TDate>() as MuiPickerFieldAdapter<TDate>;
-  if (!utils.formatTokenMap) {
-    throw new Error('This adapter is not compatible with the field components');
-  }
-  const inputRef = React.useRef<HTMLInputElement>(null);
-
-  const {
-    internalProps: {
-      value: valueProp,
-      defaultValue,
-      onChange,
-      format = utils.formats.keyboardDate,
-      readOnly = false,
-    },
-    forwardedProps: { onClick, onKeyDown, onFocus, onBlur, ...otherForwardedProps },
-    valueManager,
-    fieldValueManager,
-    validator,
-  } = params;
 
   const firstDefaultValue = React.useRef(defaultValue);
-  const focusTimeoutRef = React.useRef<NodeJS.Timeout | undefined>(undefined);
-
   const valueParsed = React.useMemo(() => {
     // TODO: Avoid this type casting, the emptyValues are both valid TDate and TInputDate
-    const value =
-      firstDefaultValue.current ?? valueProp ?? (valueManager.emptyValue as unknown as TInputValue);
+    const value = firstDefaultValue.current ?? valueProp ?? valueManager.emptyValue;
```

Comment:
Why does the default value in `firstDefaultValue.current` has priority on `valueProp`?

Reply:
Because it's a mistake :laughing:

Reply:
Fixed